android/main/java/com/couchbase/lite/internal/AndroidExecutionService.java

```diff
@@ -63,7 +62,7 @@ private CancellableTask(@NonNull Handler handler, @NonNull Runnable task) {
     // Constructor
     //---------------------------------------------
     public AndroidExecutionService() {
-        super((ThreadPoolExecutor) AsyncTask.THREAD_POOL_EXECUTOR);
+        super(new CBLExecutor());
```

Comment:
Oh, it's good to know that the AsyncTask is not deprecated!

Reply:
It is deprecated!  It is not backed by a SynchronousQueue (a queue with max size 1!).  That makes it pretty unusable (as was the intention) for people like us.